pkg/k8s/client.go

```diff
@@ -41,6 +41,11 @@ type Access struct {
 	EventRecorder record.EventRecorder
 }
 
+type NamespacedObject interface {
```

Comment:
Would it make sense to use `NamespacedName` from `apimachinery.types`:
https://godoc.org/k8s.io/apimachinery/pkg/types ?

Reply:
The problem is with `NamespacedName` is that it's a struct, and we need an interface here.

Reply:
Wondering why that's the case. 
Wouldn't this work? :
```
// IsJobFinished returns true if the given job has already succeeded or failed
func (a *Access) IsJobFinished(job NamespacedName) (finished bool, err error) {
	job, err := a.Clientset.BatchV1().Jobs(job.Namespace).Get(
		job.Name, metav1.GetOptions{})
	if err != nil {
		return false, err
	}

	finished = job.Status.Succeeded+job.Status.Failed > 0
	return finished, nil
}
```

Reply:
This way you likely cannot pass a CR object to IsJobFinished, but that's completely reasonable as CR is not a job.
Constructing the NamespacedName object from the CR would take one additional line in fio's controller.

Reply:
It is not clear how we bind GetName() and GetNamespace() functions to NamespacedObject. It is actually an [ObjectMeta](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L137-L139) which implements the [Object](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L33-L66) interface.

Reply:
@apollolemonte in Go interfaces are implemented implicitly. In this case ObjectMeta implicitly implements NamespacedObject, because it has all the 2 required methods.